developer/services/qrexec3.md

```diff
@@ -16,94 +16,77 @@ redirect_from:
 - /wiki/Qrexec3Implementation/
 ---
 
-# Command execution in VMs #
-
-(*This page is about qrexec v3. For qrexec v2, see
-[here](/doc/qrexec2/).*)
-
-The **qrexec** framework is used by core Qubes components to implement
-communication between domains. Qubes domains are isolated by design, but
-there is a need for a mechanism to allow the administrative domain (dom0) to
-force command execution in another domain (VM). For instance, when user
-selects an application from the KDE menu, it should be started in the selected
-VM. Also, it is often useful to be able to pass stdin/stdout/stderr from an
-application running in a VM to dom0 (and the other way around). In specific
-circumstances, Qubes allows VMs to be initiators of such communications (so,
-for example, a VM can notify dom0 that there are updates available for it).
-
-
-## Qrexec basics ##
-
-Qrexec is built on top of vchan (a library providing data links between
-VMs). During domain creation a process named `qrexec-daemon` is started
-in dom0, and a process named `qrexec-agent` is started in the VM. They are
-connected over **vchan** channel. `qrexec-daemon` listens for connections
-from dom0 utility named `qrexec-client`. Typically, the first thing that a
-`qrexec-client` instance does is to send a request to `qrexec-daemon` to
-start a process (let's name it `VMprocess`) with a given command line in
-a specified VM (`someVM`). `qrexec-daemon` assigns unique vchan connection
-details and sends them both to `qrexec-client` (in dom0) and `qrexec-agent`
-(in `someVM`). `qrexec-client` starts a vchan server which `qrexec-agent`
-connects to. Since then, stdin/stdout/stderr from the VMprocess is passed
-via vchan between `qrexec-agent` and the `qrexec-client` process.
-
-So, for example, executing in dom0:
+# Qrexec: command execution in VMs
 
-    qrexec-client -d someVM user:bash
+(*This page is about qrexec v3. For qrexec v2, see [here](/doc/qrexec2/).*)
+
+The **qrexec** framework is used by core Qubes components to implement communication between domains.
+Qubes domains are strictly isolated by design.
+However, the OS needs a mechanism to allow the administrative domain (dom0) to force command execution in another domain (VM).
+For instance, when a user selects an application from the KDE menu, it should start in the selected VM.
+Also, it is often useful to be able to pass stdin/stdout/stderr from an application running in a VM to dom0 (and the other way around).
+(For example, so that a VM can notify dom0 that there are updates available for it).
+By default, Qubes allows VMs initiate such communications in specific circumstances.
+The qrexec framework generalizes this process.
+It allows users and developers to use and design secure inter-VM tools.
+
+## Qrexec basics: architecture and examples
```

Comment:
We use `##` markers on both sides, see https://www.qubes-os.org/doc/doc-guidelines/#markdown-conventions

Reply:
To be fair, I've recently been taking a more relaxed approach toward this, because I realized that it's harder to edit and maintain when they're on both sides, and we don't gain much. I'll update the guidelines on this point.

Reply:
Okay, @andrewdavidwong: will leave as is.